lib/Immutable.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const util = require('util');
+
+/**
+ * Error for Immutable variables
+ *
+ * @extends Error
+ * @class Errors.Immutable
+ * */
+class Immutable extends Error{
+    /**
+ * @example
+
+      let piKindOf = 22/7;
+
+      function getItRight(){
+          if(typeof piKindOf != 'undefined'){
+              const err = new Errors.Immutable;
+              err.setMessage(
+                  'piKindOf',
+                  'global'
+              );
+              throw err;
+          }
+          piKindOf = Math.PI;
+      }
+
+      getItRight();
+
+
+ * @example
+ *
+ * ```sh
+ *
+ /git/node-error-classes/example/immutable.js:14
+     throw err;
+     ^
+
+Immutable: 'piKindOf' has been defined and cannot be changed on the scope of : 'global'.
+     Variable names here must be unique
+
+ * ```
+ *
+ * @method Errors.Immutable.setMessage
+ * @param {String} varaibleName Name of variable
+ * @param {Any} scope     Scope of varaible
+ * @returns {String}    compiled error message
+ */
+    setMessage(varaibleName, scope){
+        this.name = 'Immutable';
+        this.message = `${util.inspect(varaibleName, {colors: true})} has been defined and cannot be changed on the scope of : ${util.inspect(scope, {colors: true})}.
+        Variable names here must be unique`;
```

Comment:
I believe in all of the other error classes the "scope" is called like fromVar or something. Maybe they should all be called scope? Seems like a better term.